ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayloadV4.java

```diff
@@ -63,8 +63,6 @@ protected ValidationResult<RpcErrorType> validateParameters(
     } else if (maybeBeaconBlockRootParam.isEmpty()) {
       return ValidationResult.invalid(
           RpcErrorType.INVALID_PARAMS, "Missing parent beacon block root field");
-    } else if (payloadParameter.getDeposits() == null) {
```

Comment:
We don't need this early check as this is already handled by the `DepositValidator` on `AbstractEngineNewPayload`. However, if we do want to have this early check, we would need to modify some tests to support it.

IMHO, this comes down to: do we ever have a scenario where GetPayloadV4 is called pre-Prague?

Reply:
Agree it appears to be a duplicate check. If we are having to validate params for other fields here then think it makes sense to do it all in one place, rather than spread the check out.

Feels like the domain-level DepositsValidator shouldn't be doing this check if we already have this here

Reply:
We have shared logic between the different version of `engine_newPayload` implemented on `AbstractEngineNewPayload`, because of that we still need the "conditional" validation on `AbstractEngineNewPayload`, determined by protocol schedule + DepositsValidator (choosing a validator implementation that makes sense for what fork we are in).

When it comes to `EngineNewPayloadV4#validateParameters`, we could enforce checking that the deposits field isn't empty, assuming that we only use V4 after Prague (need to confirm this).

Depending on what we think is the correct thing to do, we will need to adjust some of the testing because as of now, `EnginePayloadV4` test extends `EnginePayloadV3` and etc,  and a lot of these tests will fail.

Reply:
Ok this method should not be used pre-Prague. I'll adjust a few things.

Reply:
@jflo might have the most recent context about the best way to migrate to V4.